src/pages/Home/Home.page.jsx

```diff
@@ -1,29 +1,54 @@
-import React from 'react';
+import React, { useState } from 'react';
 import Header from '../../components/Header';
 import VideoCard from '../../components/VideoCard';
-
+import VideoDetail from '../VideoDetail';
 import { GridContainer, Title } from './styled';
-import { items } from '../../mock/youtube-videos-mock.json';
+
+import { useVideosSearch } from '../../hooks/useVideosSearch';
 
 const HomePage = () => {
-  return (
-    <React.Fragment>
-      <Header />
-      <Title>Welcome to my minichallenge :)</Title>
-      <GridContainer>
-        {items.map((i) => {
-          const { title, description, thumbnails } = i.snippet;
-          return (
-            <VideoCard
-              title={title}
-              description={description}
-              thumbnail={thumbnails.high.url}
-            />
-          );
-        })}
-      </GridContainer>
-    </React.Fragment>
-  );
+  const [keyword, setKeyword] = useState('wizeline');
+  const [selectedVideo, setSelectedVideo] = useState();
+  const videos = useVideosSearch(keyword);
+
+  const onSubmit = (newKeyword) => {
+    console.log(newKeyword);
+    setKeyword(newKeyword);
+  };
+
+  let content;
+  if (!selectedVideo) {
+    content = (
+      <React.Fragment>
+        <Header onSubmit={onSubmit} />
+        <Title>Welcome to my minichallenge :)</Title>
+        <GridContainer>
+          {videos.map((i) => {
+            const { title, description, thumbnails } = i.snippet;
+            return (
+              <VideoCard
+                title={title}
+                onClick={() => {
+                  setSelectedVideo(i);
+                }}
+                description={description}
+                thumbnail={thumbnails.high.url}
+              />
+            );
+          })}
+        </GridContainer>
+      </React.Fragment>
+    );
+  } else {
```

Comment:
Here you have an `if/else` statement to check if the user is in the home page or in a specific video, with `Routing` you can use a route switch to handle this logic, but it is something that will be part of the next topics